backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +72,97 @@ def disambiguate_signatures(self, clusters):
+        author_reference_url = matched_authors_refs[0]
+        return {"$ref": author_reference_url}
+
+
+def assign_reference_to_author(author, updated_authors, reference):
+    author["record"] = reference
+    updated_authors.append(reference["$ref"].split("/")[-1])
+
+
+def match_literature_author_with_configs(author, updated_authors, record):
+    configs = [
+        current_app.config["AUTHOR_MATCHER_NAME_CONFIG"],
+        current_app.config["AUTHOR_MATCHER_NAME_INITIALS_CONFIG"],
+        current_app.config["AUTHOR_MATCHER_COLLABORATION_CONFIG"],
+        current_app.config["AUTHOR_MATCHER_AFFILIATION_CONFIG"],
+    ]
+
+    parsed_name = ParsedName.loads(author.get("full_name"))
+    author_matcher_data = {
+        "first_name": parsed_name.first,
+        "last_name": parsed_name.last,
+        "full_name": author.get("full_name"),
+        "collaborations": get_value(record, "collaborations.value", []),
+        "affiliations": get_value(author, "affiliations.value", []),
+    }
+
+    for config in configs:
+        matched_recid = match_literature_author(author_matcher_data, config)
+        if matched_recid:
+            assign_reference_to_author(author, updated_authors, matched_recid)
+            return author
+
+
 @shared_task(ignore_result=False, bind=True)
 def disambiguate_authors(self, record_uuid):
     record = InspireRecord.get_record(record_uuid)
     if "Literature" not in record["_collections"]:
         return
     authors = record.get_modified_authors()
     updated_authors = []
+
     for author in authors:
         if author.get("curated_relation"):
             continue
-        matched_authors = [
+        matched_authors = {
             matched_record["_source"]["control_number"]
             for matched_record in match(
                 author, current_app.config["AUTHOR_MATCHER_EXACT_CONFIG"]
             )
-        ]
-        if len(set(matched_authors)) == 1:
-            author_control_number = matched_authors[0]
-            author["record"] = get_ref_from_pid("aut", author_control_number)
-            updated_authors.append(str(author_control_number))
+        }
+        if len(matched_authors) == 1:
+            author_control_number = matched_authors.pop()
+            assign_reference_to_author(
+                author,
+                updated_authors,
+                get_record_ref(author_control_number, endpoint="authors"),
+            )
+            continue
+        match_literature_author_with_configs(author, updated_authors, record)
+        if not author.get("record"):
+            new_author_data = {
+                "name": {"value": author["full_name"]},
+                "_private_notes": [
+                    {
+                        "source": "literature_record",
+                        "value": str(record.get("control_number")),
+                    }
+                ],
+            }
+            new_author = create_new_empty_author(**new_author_data)
```

Comment:
the name is a misnomer now. Plus I don't think you need to do this, you could do
```suggestion
            new_author = create_new_empty_author()
            new_author.update(new_author_data)
```